recipes/build123d/build.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+export SETUPTOOLS_SCM_PRETEND_VERSION="$PKG_VERSION"
```

Comment:
Instead of this file can you try using:

```yaml
  script: SETUPTOOLS_SCM_PRETEND_VERSION={{ version }} {{ PYTHON }} -m pip install . -vv
``` 

in the meta.yaml file `build` section?

Reply:
You can actually use [build: script_env:](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#use-environment-variables) and an earlier revision did that, and that worked, but the upstream [build123d/version.py](https://github.com/gumyr/build123d/blob/235260a8000fa9d5174df0efea100d7cff332ec6/src/build123d/version.py) doesn't single-source from package metadata and expects it's own hard-copy, which is why I needed the other script line to write the version file.

I could patch that but was trying to steer away from just patching all my problems away.

Reply:
That said I'm happy to change if I erred on the wrong side of caution there

Reply:
> I could patch that but was trying to steer away from just patching all my problems away.

No need to patch but you can open an issue upstream to avoid the hard-copy of that file. It kinds of defeats the purpose.

Reply:
Yep, good idea, I have a couple of things to upstream, including allowing either lib3mf. Thanks!